MagicalCryptoWallet/Services/IndexBuilderService.cs

```diff
@@ -24,7 +24,67 @@ public class IndexBuilderService
 		private List<FilterModel> Index { get; }
 		private AsyncLock IndexLock { get; }
 
-		private Dictionary<OutPoint, Script> Bech32UtxoSet { get; }
+		private Dictionary<OutPoint, Script> Bech32UtxoSet { get; } //Change to observable concurrent dictionary? https://www.nuget.org/packages/ParallelExtensionsExtras/
+
+		
+
+		internal static class ActionHistoryHelper
```

Comment:
Could you make it non-static? While it'll be a singleton for now, later or during tests it could cause problems.

Reply:
This is a very special class only for this special purpose. I do not want anybody to create more instances or make it visible from outside. 
Of course I can change it to non static.

Reply:
Please make it. My concern is if `IndexBuilderService` becomes used in different places it'll break. In fact it probably will while testing. Tests are usually running concurrently, and back then my static global classes were the cause of my unreliable tests, so I'd like to avoid it this time.

> I do not want anybody to create more instances or make it visible from outside.

I guess you can make it private, too. If you don't want to write tests for it. I think tests are not needed.